specification/ai/OpenAI.Assistants/assistants/models.tsp

```diff
@@ -0,0 +1,147 @@
+  @doc("The description of the assistant.")
+  description: string | null;
+
+  @doc("The ID of the model to use.")
+  `model`: string;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The system instructions for the assistant to use.")
+  instructions: string | null;
+
+  @doc("The collection of tools enabled for the assistant.")
+  tools: ToolDefinition[] = [];
+
+  @projectedName("json", "file_ids")
+  @doc("A list of attached file IDs, ordered by creation date in ascending order.")
+  fileIds: string[] = [];
+
+  ...RequiredNullableMetadata;
+}
+
+@doc("The request details to use when creating a new assistant.")
+@added(ServiceApiVersions.v2024_02_15_preview)
+model AssistantCreationOptions {
+  @doc("The ID of the model to use.")
+  `model`: string;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The name of the new assistant.")
+  name?: string | null;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The description of the new assistant.")
+  description?: string | null;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The system instructions for the new assistant to use.")
+  instructions?: string | null;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The collection of tools to enable for the new assistant.")
+  tools?: ToolDefinition[] = [];
+
+  @projectedName("json", "file_ids")
+  @doc("A list of previously uploaded file IDs to attach to the assistant.")
+  fileIds?: string[] = [];
+
+  ...OptionalNullableMetadata;
+}
+
+@doc("The request details to use when modifying an existing assistant.")
+@added(ServiceApiVersions.v2024_02_15_preview)
+model UpdateAssistantOptions {
+  @path
+  @doc("The ID of the assistant to modify.")
+  assistantId: string;
+
+  @doc("The ID of the model to use.")
+  `model`?: string;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The modified name for the assistant to use.")
+  name?: string | null;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The modified description for the assistant to use.")
+  description?: string | null;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "OpenAI uses explicit nullability, distinct from optionality"
+  @doc("The modified system instructions for the new assistant to use.")
+  instructions?: string | null;
+
+  @doc("The modified collection of tools to enable for the assistant.")
+  tools?: ToolDefinition[] = [];
```

Comment:
Is this really true? I have to repeat the tool definitions or they will be replaced with an empty list if I want to change the description of the assistant?

Reply:
I believe that may be correct, yes. `ModifyAssistant` seems to function as a "replace" or "recreate" rather than a patch. This isn't abundantly clear in OpenAI's docs and we should be sure to validate and clarify that ourselves. This was a factor in us standardizing on "Update" vs. "Modify" despite the divergence.

- [API reference](https://platform.openai.com/docs/api-reference/assistants/modifyAssistant)
- [OpenAPI](https://github.com/openai/openai-openapi/blob/97dfb59c89ce0550e9360a7d437491d556e5a65e/openapi.yaml#L7413)

Reply:
Will continue to evaluate this and follow up; at the least, we need to clarify and document this really well.